app/controllers/TicketBooking.scala

```diff
@@ -0,0 +1,204 @@
+package controllers
+
+import models._
+import play.api.data.Forms._
+import play.api.data._
+import play.api.mvc._
+import play.filters.csrf.CSRF
+
+import java.time.LocalDateTime
+import javax.inject._
+import scala.util.{Failure, Success, Try}
+
+@Singleton
+class TicketBooking @Inject() (cc: MessagesControllerComponents) extends MessagesAbstractController(cc) {
+
+  private val availableScreenings: List[Screening] = DefaultScreenings.getAvailableScreenings
+  private var availableScreeningsFiltered: List[Screening] = Nil
+  private var chosenScreeningOption: Option[Screening] = None
+  private var chosenSeats: Seq[(Int, Int)] = Nil
+  private var chosenTickets: Seq[Ticket] = Nil
+
+  val customerDataForm: Form[Customer] = Form(mapping(
+    "name" -> text(3),
+    "surname" -> text(3)
+  )(Customer.apply)(Customer.unapply))
+
+  def ticketBookingInit: Action[AnyContent] = Action { implicit request =>
+    val csrfToken = CSRF.getToken.get.value
+    Ok(views.html.ticketBookingInit()).withSession("csrfToken" -> csrfToken)
+  }
+
+  def chooseDateAndTime: Action[AnyContent] = Action { implicit request =>
+    val postVals = request.body.asFormUrlEncoded
+    postVals.map { args =>
+      if (args("dateTime").head.nonEmpty) {
+        val chosenDateTime = args("dateTime").head
+        println(chosenDateTime)
+        Redirect(routes.TicketBooking.availableMovies).withSession(request.session.+("dateTime" -> chosenDateTime))
+      } else {
+        println("empty date time")
+        Redirect(routes.TicketBooking.ticketBookingInit).flashing("error" -> "choose date and time!")
+      }
+    }.getOrElse(Redirect(routes.TicketBooking.ticketBookingInit))
+  }
+
+  def availableMovies: Action[AnyContent] = Action { implicit request =>
+    val chosenDateTimeTry = Try(LocalDateTime.parse(request.session.get("dateTime").get))
+    chosenDateTimeTry match {
+      case Failure(_) =>
+        Redirect(routes.TicketBooking.ticketBookingInit)
+          .withNewSession.flashing("error" -> "wrong date-time format")
+      case Success(dateTime) =>
+        availableScreeningsFiltered =
+          availableScreenings.filter(screening =>
+            screening.startTime.plusMinutes(90).isAfter(dateTime) &&
+              screening.startTime.minusHours(3).isBefore(dateTime) &&
+              screening.startTime.minusMinutes(15).isAfter(LocalDateTime.now())
+          )
+        Ok(views.html.availableMovies(availableScreeningsFiltered)).withSession(request.session)
+    }
+  }
+
+  def chooseScreening: Action[AnyContent] = Action { implicit request =>
+    val postVals = request.body.asFormUrlEncoded
+    postVals.map { args =>
+      val title = args("title").head
+      val startTime = args("startTime").head
+      chosenScreeningOption = availableScreenings
+        .find(scr => scr.movie.title == title && scr.startTime == LocalDateTime.parse(startTime))
```

Comment:
why LocalDateTime may be not the best choice for a real app?

Reply:
Probably there could be some issues with different timezones - f.e. timezone on the JVM differs from the timezone where the Cinema is placed or JVM timezone differs from the timezone of the database or some problems like that.